src/OpenTelemetry/Logs/Pool/LogRecordPool.cs

```diff
@@ -0,0 +1,39 @@
+// <copyright file="LogRecordPool.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#nullable enable
+
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Logs
+{
+    /// <summary>
+    /// Manages a pool of <see cref="LogRecord"/> instances.
+    /// </summary>
+    public static class LogRecordPool
+    {
+        /// <summary>
+        /// Resize the pool.
+        /// </summary>
+        /// <param name="capacity">The maximum number of <see cref="LogRecord"/>s to store in the pool.</param>
+        public static void Resize(int capacity)
```

Comment:
Is it okay to expose a public API that could allocate a huge array at runtime post-startup?

Reply:
🤷 You have an alternative in mind?

I added this because I thought there might be users interested in fine-tuning the pool. But we could make it internal initially and see if anyone asks. I'm surprised not more people commenting on it 😄

Reply:
It'd be good to start without exposing this publicly.

If there is an ask (eg: https://github.com/dotnet/runtime/issues/65264), we can expose it then.

Reply:
Following the general principle of not exposing anything until there is a spec, or super strong need - lets keep this internal to begin with.

Reply:
Updated